tools/flow9/backends/cpp/cpp.flow

```diff
@@ -445,3 +482,42 @@ switchType2caseIndexes(switchType : BType, backend : Backend<CppInfo>) -> Tree<s
 		BTypeFunction(__, __) : makeTree(); // error
 	}
 }
+
+isBTypeUnionType(type : BType, backend : Backend<CppInfo>) -> bool {
+	switch (type : BType) {
+		BTypePar(id) : containsKeyTree(backend.info.unionTypes, id);
+		BTypeName(id, __) : containsKeyTree(backend.info.unionTypes, id);
+		BTypeFunction(__, __) : false;
+	}
+}
+
+//up cast
+btype2superType(resultType : BType, type : BType, backend : Backend<CppInfo>) -> Maybe<string> {
```

Comment:
Explain what the args are. It is not clear to me which is the subtype (struct) and which is the supertype (union).